src/Term/TermList.php

```diff
@@ -199,9 +193,14 @@ public function clear() {
 	 * @since 8.1
 	 *
 	 * @param iterable|Term[] $terms
+	 * @throws InvalidArgumentException
 	 */
 	public function addAll( /* iterable */ $terms ) {
 		foreach ( $terms as $term ) {
+			if ( !( $term instanceof Term ) ) {
+				throw new InvalidArgumentException( 'Every element in $terms must be an instance of Term' );
```

Comment:
Not sure if this is by (some) design but neither the constructor did have nor `addAll()` now appear to be having a test case that asserts the `InvalidArgumentException`.

Reply:
Good catch, added those